main/tests.py

```diff
@@ -87,3 +88,44 @@ def test_unlogged_user_access_to_homepage(self, client):
         response = client.get("/homepage/")
         assert response.status_code == 302
         assert response['Location'] == '/login/?next=/homepage/'
+
+
+@pytest.mark.django_db
+class TestDogOwnerHomePageView:
+    def test_dog_owner_present_all_day_cares_at_entrypoint(self, client, create_dog_owner_user):
+        client.force_login(user=create_dog_owner_user.user)
+        response = client.get("/homepage/")
+        day_care_queryset = response.context['day_care_queryset']
+        assert day_care_queryset.count() == DayCare.objects.all().count()
+
+    def test_successful_dog_owner_search_for_day_care(self, client, create_dog_owner_user):
+        client.force_login(user=create_dog_owner_user.user)
+        search_form = {'area': "center",
+                       'city': "tel aviv",
+                       'price_per_day': 800,
+                       'name': "",
+                       'start_date': "25/05/2022",
+                       'end_date': "28/05/2022",
+                       }
+        response = client.post('/homepage/', search_form, follow=True)
+        day_care_queryset = response.context['day_care_queryset']
+        for day_care in DayCare.objects.all():
+            if day_care.price_per_day <= 800 and day_care.area.lower() == 'center'\
+                    and day_care.city.lower() == 'tel aviv':
+                assert day_care in day_care_queryset
```

Comment:
maybe also check that all day_care_queryset is in filter range, 
your test checks that all filtered daycares are shown, maybe add a test that all daycares in day_care_queryset are the correct ones.
Easy hit append all daycares in the for to list, then assert equality between this list to day_care_queryset

Reply:
You right . I guess we can use a counter in the for loop instead and to check in the end if counter = day_care_queryset.count.